api/lib/validate.js

```diff
@@ -166,8 +178,95 @@ function validateOrderbook(orderbook) {
   }
 }
 
+function validateSettings(settings) {
+  if (typeof settings !== 'object') {
+    throw error('Invalid parameter: settings');
```

Comment:
Which client? The rest client is already handling them currently. A javascript client would handle them with try/catch or wrapping in promises. The server is already getting an error object (the exception).